### DISCUSSION TITLE
helm install minio with directpv failed

### DISCUSSION THREAD

**Discussion by lordk911** [minio/minio#14765] [Q&A] [answered]
hi,all:
I want to use helm install minio , first I install directpv by following the docs : https://github.com/minio/directpv/blob/master/docs/installation.md
I checked the installation by run : kubectl directpv info , the show is :
```
[root@dg-poc-master-01 ~]# kubectl directpv info
 NODE                CAPACITY  ALLOCATED  VOLUMES  DRIVES 
 • dg-poc-worker-01  2.0 TiB   0 B        0        1      
 • dg-poc-worker-02  2.0 TiB   0 B        0        1      
```
and then I try to install minio by run : helm install --set rootUser=rootuser,rootPassword=rootpass123,replicas=2,persistence.storageClass=directpv  --generate-name minio/minio
but the pvc keep in pending , when I describe it , shows :
```
Events:
  Type     Reason              Age                    From                         Message
  ----     ------              ----                   ----                         -------
  Warning  ProvisioningFailed  114s (x26 over 7m55s)  persistentvolume-controller  [storageclass.storage.k8s.io](http://storageclass.storage.k8s.io/) "directpv" not found
```

**Comment by lordk911** [comment, marked as answer]
storageClass name should use directpv-min-io

**Comment by edqallen** [comment]
Has anything changed since this issue was opened and answered?  I'm having the same problem, and explicitly setting the storageClass is not resolving the issue.  Using the following command, with or without the storageClass name set, results in 16 pods being created and the following message from each of them: `0/6 nodes are available: 6 pod has unbound immediate PersistentVolumeClaims.`

```
kubectl directpv drives ls                
 DRIVE     CAPACITY  ALLOCATED  FILESYSTEM  VOLUMES  NODE        ACCESS-TIER  STATUS   
 /dev/sdb  32 GiB    -          xfs         -        c02-work01  -            Ready    
 /dev/sdb  32 GiB    -          xfs         -        c02-work02  -            Ready    
 /dev/sdb  32 GiB    -          xfs         -        c02-work03  -            Ready    
```

```
kubectl directpv info
 NODE          CAPACITY  ALLOCATED  VOLUMES  DRIVES 
 • c02-work01  32 GiB    0 B        0        1      
 • c02-work02  32 GiB    0 B        0        1      
 • c02-work03  32 GiB    0 B        0        1      
```

The first helm install attempt used just this command, trusting the defaults would be sane and functional:

`helm install minio minio/minio --namespace=minio --create-namespace`

Followed by: `helm install minio minio/minio --set persistence.storageClass=directpv-min-io --namespace=minio --create-namespace`

TIA for any insight/assistance.

**Reply by harshavardhana** [reply]
> Has anything changed since this issue was opened and answered? I'm having the same problem, and explicitly setting the storageClass is not resolving the issue. Using the following command, with or without the storageClass name set, results in 16 pods being created and the following message from each of them: 0/6 nodes are available: 6 pod has unbound immediate PersistentVolumeClaims.

6 nodes - 3 drives how will it work? only 3 nodes? by default helm chart asks for 500Gi and your disks are just 32Gi
